tokenmagic/module/tokenmagic.js

```diff
@@ -284,6 +284,36 @@ export function getPlaceableById(id, type) {
   return placeable;
 }
 
+/**
+ * Randomizes params using 'randomized' field. 
+ * 'randomized' is an object consisting of keys named after params to be randomized, which map either to arrays or ranges
+ * which will be used to generate a random value.
+ * e.g.
+ * {
+ *  param1: ['foo1', 'foo2', 'foo3'],
+ *  param2: { list: ['foo1', 'foo2', 'foo3'], link: 'param5'},
+ *  param3: { val1: 0, val2: 1, step: 0.1},
+ *  param4: { val1: 0, val2: 10, step: 1, link: 'param6'},
+ * }
+ * 'link' will assign the same generated value to one other param.
+ */
+async function randomizeParams(params) {
+  for(const [param, opts] of Object.entries(params.randomized)){
+    if(Array.isArray(opts) || opts.hasOwnProperty('list')){
+      const list = opts.list ?? opts;
+      params[param] = list[Math.floor(Math.random() * list.length)];
+    } else {
+      const min = Math.min(opts.val1, opts.val2);
+      const max = Math.max(opts.val1, opts.val2);
+      const stepsInRange = (max - min + (Number.isInteger(opts.step) ? 1 : 0)) / opts.step;
+      params[param] = Math.floor(Math.random() * stepsInRange) * opts.step + min;
+    }
+    if (opts.hasOwnProperty('link')) {

```

Comment:
We should avoid here a division by 0 by checking `opts.step`.

Reply:
Added a check to default `opts.step` to 1 if it's falsy. It only makes sense for step to be >=0